pkg/interactor/data.go

```diff
@@ -44,6 +44,16 @@ func (d *DataInteractor) Create(data Data) error {
 	return err
 }
 
+func (d *DataInteractor) DeleteByDevice(deviceId string) (error error) {
```

Comment:
It's advisable to create one interactor file for each use case to avoid mixing up the business logic. In this case, you should create another file named as `delete_data.go`.  One possibility is to create a DeleteData interactor (with only one method to be externally called - `DeleteData()`) that receives the `deviceId` option so that it can decide to remove all or by `deviceId`.

Reply:
Agree with @joaoaneto! As these files will be in same package, you can just move the code around those files. You can also create a file called `interactor.go` to hold the base code, as the `DataInteractor` struct definition and `NewDataInteractor` constructor method. This is what we do in [`babeltower` interactor package](https://github.com/CESARBR/knot-babeltower/tree/master/pkg/thing/interactors).

Let us knot if you have some questions or suggestions! :100:

Reply:
As discussed in the `slack` channel, I'd prefer you add the delete in another file `delete_data.go` but without following the babeltower pattern.